src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -832,14 +838,20 @@ private static FormulaValue LeftOrRight(IRContext irContext, FormulaValue[] args
                 throw new NotImplementedException("Should have been handled by IR");
             }
 
-            return new StringValue(irContext, leftOrRight(source.Value, (int)count.Value));
+            TryGetIntValue(count.Value, out int cnt);
+
+            return new StringValue(irContext, leftOrRight(source.Value, cnt));
         }
 
         private static FormulaValue Find(IRContext irContext, FormulaValue[] args)
         {
             var findText = (StringValue)args[0];
             var withinText = (StringValue)args[1];
-            var startIndexValue = (int)((NumberValue)args[2]).Value;
+
+            if (!TryGetIntValue(((NumberValue)args[2]).Value, out int startIndexValue))
```

Comment:
>((NumberValue)args[2]).Value [](http://example.com/codeflow?start=32&length=28)

Could we pass DecimalValue in here?

If so, we could have an overload for TryGetIntValue that accepts FormulaValue and handles both Decimal and Number values (and fails on anything else)

Reply:
updated trygetint.